lib/dynhds.c

```diff
@@ -365,3 +369,26 @@ CURLcode Curl_dynhds_h1_dprint(struct dynhds *dynhds, struct dynbuf *dbuf)
   return result;
 }
 
+#ifdef USE_NGHTTP2
+
+nghttp2_nv *Curl_dynhds_to_nva(struct dynhds *dynhds, size_t *pcount)
+{
+  nghttp2_nv *nva = calloc(1, sizeof(nghttp2_nv) * dynhds->hds_len);
+  size_t i;
+
+  *pcount = 0;
+  if(!nva)
+    return NULL;
+
+  for(i = 0; i < dynhds->hds_len; ++i) {
+    nva[i].name = (unsigned char *)dynhds->hds[i]->name;
+    nva[i].namelen = dynhds->hds[i]->namelen;
+    nva[i].value = (unsigned char *)dynhds->hds[i]->value;
+    nva[i].valuelen = dynhds->hds[i]->valuelen;
+    nva[i].flags = NGHTTP2_NV_FLAG_NONE;
```

Comment:
What do you say about this to increase the readability a bit?

```suggestion
    struct dynhds_entry *e = dynhds->hds[i];
    DEBUGASSERT(e);
    nva[i].name = (unsigned char *)e->name;
    nva[i].namelen = e->namelen;
    nva[i].value = (unsigned char *)e->value;
    nva[i].valuelen = e->valuelen;
    nva[i].flags = NGHTTP2_NV_FLAG_NONE;
```

Reply:
updated